rnd/autogpt_server/autogpt_server/blocks/basic.py

```diff
@@ -1,26 +1,33 @@
-# type: ignore
-
 from autogpt_server.data.block import Block, BlockSchema, BlockOutput
 
+from typing import Any
+
 
-class ParrotBlock(Block):
+class ConstantBlock(Block):
```

Comment:
The common use-case is just relaying the input as the output, yes. This can be perceived as a constant value where you provide an execution input to the block and use the output in many blocks.

But the actual intention is providing a constant value in a stateless manner, by keep feeding the `output` data back to the `data` pin.  I have added more description as a class doc, let me know if it clears things up.

Reply:
Ok I see, now this sounds like `CacheBlock` or something. I imagine const block providing a constant value without any inputs even.
Can we just *hide* that extra input and output and link it automatically? If this can be implemented at block level that would be perfect.

Reply:
I don't think this can be implemented at the block level because the block is not a node yet.

Though this can be achieved once we added subgraph as a node capability. What you proposed is inherently already a graph, not a node.

Reply:
I have logged the subgraph support as one of our nextstep.

> Ok I see, now this sounds like CacheBlock or something.
I see, since this is not solely for 'caching' but also providing a hardcoded input value, I'll rename this to `ValueBlock`